napari/layers/shapes/shapes.py

```diff
@@ -1366,6 +1366,9 @@ def add(
             edge_width = self.current_edge_width
 
         n_new_shapes = len(data)
+        if np.array(data[0]).ndim == 1:
+            # If a single array for a shape
+            n_new_shapes = 1
```

Comment:
Can we change this to the syntax below where we have an `if` / `else` rather than overwriting the `n_new_shapes` variables. I find overwriting variables can sometimes make code harder to follow.

```suggestion
        if np.array(data[0]).ndim == 1:
            # If a single array for a shape
            n_new_shapes = 1
        else:
            n_new_shapes = len(data)
```

Reply:
not sure if I have the write number of ` ` in the suggestion (syntax coloring looks off), but I'm sure you get the proposed change